app-directory/redirect-with-fallback/redirects.d.ts

```diff
@@ -0,0 +1,9 @@
+export interface Redirect {
+  source: string
+  destination: string
+  permanent?: boolean
+  statusCode?: number
+}
+
+export const simpleRedirects: Redirect[]
+export const complexRedirects: Redirect[]
```

Comment:
Could we move this to JSDoc? consider it optional but I think it makes more sense for this case.

Reply:
Tried using JSDoc but other ts files were not able to infer the exported types:
```
app/[...slug]/route.ts:1:10 - error TS7034: Variable 'simpleRedirects' implicitly has type 'any[]' in some locations where its type cannot be determined.

1 import { simpleRedirects } from '../../redirects'
```
I've removed `redirects.d.ts` and added JSDoc comments in `redirects.js`
```js
/** @typedef {{
  source: string
  destination: string
  permanent?: boolean
  statusCode?: number
}} Redirect */

/** @type {Redirect[]} */
const simpleRedirects = []
/** @type {Redirect[]} */
const complexRedirects = []
```